### PR TITLE
Inc objective for read-only requests in "namespace" scope: from 5s to 30s

### PR DESCRIPTION
**Pull request by azylinski**
Historically, threshold for namespace-scoped LISTs was 5s. It was based on the assumption that a namespace is limited in size [1], but that ended up being invalid. There were clusters with (small, but still) tens of thousands of objects of a given type in a namespace.

[1] https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md

/assign wojtek-t

### CODE REVIEW

File: sig-scalability/slos/api_call_latency.md [hunk 1]
```diff
@@ -5,23 +5,24 @@
...
-| __Official__ | Latency<sup>[1](#footnote1)</sup> of non-streaming read-only<sup>[3](#footnote3)</sup> API calls for every (resource, scope<sup>[4](#footnote4)</sup>) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, scope) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day (a) <= 1s if `scope=resource` (b) <= 5s if `scope=namespace` (c) <= 30s if `scope=cluster` |
+| __Official__ | Latency<sup>[1](#footnote1)</sup> of non-streaming read-only<sup>[3](#footnote3)</sup> API calls for every (resource, scope<sup>[4](#footnote4)</sup>) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, scope) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day: (a) <= 1s if `scope=resource` (b) <= 30s otherwise (if `scope=namespace` or `scope=cluster`) |
```
Comment (wojtek-t) [inline sig-scalability/slos/api_call_latency.md:8]: We need a footnote describing that historically we had 5s for namespaces.

And describe why we changed that (basically the fact that it was assuming some limits on objects within a single namespace, which given evolution of k8s clearly isn't the case.

File: sig-scalability/slos/api_call_latency.md [hunk 2]
```diff
@@ -5,23 +5,28 @@
...
-from a single namespace or (c) `cluster` if it spawns objects from multiple
-namespaces.
+<a name="footnote4">\[4\]</a> A scope of a request can be either
+- `resource` - if the request is about a single object
+- `namespace` - if it is about objects from a single namespace
+- `cluster` - if it spawns objects from multiple namespaces
+
+<a name="footnote5">\[5\]</a> Historically, the threshold for LISTs in
+`scope=namespace` was set to 5 seconds. It was assuming limitted size on
+objects within a single namespace. Given k8s evolution, that is not the case.
```
Comment (wojtek-t) [inline sig-scalability/slos/api_call_latency.md:28]: Let's change to something like:

Historically, the threshold for LISTs with `scope=namespace` was set to 5 seconds. However, the threshold was chosen when Kubernetes doesn't support the scale it supports today and when individual namespace didn't contain tens of thousands (if not more) of objects of a given type. We adjusted the limits to accommodate the 
usage patterns change, given that users are fine with listing tens of thousands of objects taking more than 5 seconds.

**Comment by wojtek-t** [comment]
@kubernetes/sig-scalability  @marseel @shyamjvs  - FYI

/lgtm
/approve

**Comment by dgrisonnet** [comment]
@azylinski based on this change, does it still even make sense to mention limits per namespace in https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md?

**Comment by wojtek-t** [comment]
> @azylinski based on this change, does it still even make sense to mention limits per namespace in https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md?

It does - because the limit is usually not an implication of API call SLO, but rather something else (e.g. how many entries we can have in iptables, how much a controller can process items, etc.)

**Comment by dgrisonnet** [comment]
Makes sense, thank you for the clarification :)

**Pull request merged**